File: backend/models/task_assignee.go
package models

import "time"

// TaskAssignee 任务开发人员关联表
// 支持一个任务分配多名开发，每人独立跟踪开发状态
type TaskAssignee struct {
	ID        uint      `json:"id" gorm:"primaryKey"`
	TaskID    uint      `json:"task_id" gorm:"not null;index"`
	UserID    uint      `json:"user_id" gorm:"not null;index"`
	User      User      `json:"user,omitempty" gorm:"foreignKey:UserID"`
	Platform  string    `json:"platform"` // iOS / Android / 后台 / 前端 / 其他
	Status    string    `json:"status" gorm:"default:'pending'"` // pending / developing / developed
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: backend-java/src/main/java/com/management/task/entity/TaskAssignee.java
package com.management.task.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.management.user.entity.User;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.time.LocalDateTime;

@Schema(description = "任务指派人")
@Data
@TableName("task_assignees")
public class TaskAssignee {
    /** 记录ID */
    @TableId(type = IdType.AUTO)
    private Long id;
    /** 任务ID */
    private Long taskId;
    /** 用户ID */
    private Long userId;
    /** 关联用户对象（非数据库字段） */
    @TableField(exist = false)
    private User user;
    /** 平台信息 */
    private String platform;
    /** 分配状态 */
    private String status;
    /** 创建时间 */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    /** 更新时间 */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
